akka-http-core/src/test/scala/akka/http/javadsl/HttpExtensionApiSpec.scala

```diff
@@ -0,0 +1,393 @@
+
+      val pair: Pair[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]] =
+        Source.single(new Pair(HttpRequest.GET("/"), NotUsed.getInstance()))
+          .viaMat(poolFlow, Keep.right[NotUsed, HostConnectionPool])
+          .toMat(
+            Sink.head(),
+            Keep.both[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]])
+          .run(materializer)
+
+      val response = pair.second.toCompletableFuture.get(1, TimeUnit.SECONDS)
+    }
+
+    "create a cached connection pool 3" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val poolFlow: Flow[Pair[HttpRequest, NotUsed], Pair[Try[HttpResponse], NotUsed], HostConnectionPool] =
+        http.cachedHostConnectionPool[NotUsed](s"$host:$port", materializer)
+
+      val pair: Pair[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]] =
+        Source.single(new Pair(HttpRequest.GET("/"), NotUsed.getInstance()))
+          .viaMat(poolFlow, Keep.right[NotUsed, HostConnectionPool])
+          .toMat(
+            Sink.head(),
+            Keep.both[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]])
+          .run(materializer)
+
+      val response = pair.second.toCompletableFuture.get(1, TimeUnit.SECONDS)
+    }
+
+    "create a host connection pool 1" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val poolFlow = http.newHostConnectionPool[NotUsed](ConnectHttp.toHost(host, port), materializer)
+
+      val pair: Pair[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]] =
+        Source.single(new Pair(HttpRequest.GET("/"), NotUsed.getInstance()))
+          .viaMat(poolFlow, Keep.right[NotUsed, HostConnectionPool])
+          .toMat(
+            Sink.head(),
+            Keep.both[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]])
+          .run(materializer)
+
+      val response = pair.second.toCompletableFuture.get(1, TimeUnit.SECONDS)
+    }
+
+    "create a host connection pool 2" in {
+      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
+      val poolFlow = http.newHostConnectionPool[NotUsed](ConnectHttp.toHost(host, port), poolSettings, loggingAdapter, materializer)
+
+      val pair: Pair[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]] =
+        Source.single(new Pair(HttpRequest.GET("/"), NotUsed.getInstance()))
+          .viaMat(poolFlow, Keep.right[NotUsed, HostConnectionPool])
+          .toMat(
+            Sink.head(),
+            Keep.both[HostConnectionPool, CompletionStage[Pair[Try[HttpResponse], NotUsed]]])
+          .run(materializer)
+
+      val response = pair.second.toCompletableFuture.get(1, TimeUnit.SECONDS)
+    }
+
+    "allow access to the default client https context" in {
+      http.defaultClientHttpsContext.isSecure should not equal (null)
```

Comment:
this (and the next) should probably test against boolean values, which serves the double purpose of not switching between different settings inadvertently